### PR TITLE
Big bunch of minor cleanups

### PR DESCRIPTION
**Pull request by matozoid**
Lots of small things that were annoying me for a while now. Most of it refactored with tools.

Also closes this: https://github.com/javaparser/javaparser/issues/245

### CODE REVIEW

**Comment by matozoid** [comment]
https://github.com/javaparser/javaparser/issues/246 could be closed because none of that type of error are reported.

File: javaparser-core/src/main/java/com/github/javaparser/PositionUtils.java [hunk 1]
```diff
@@ -132,8 +132,7 @@ public static boolean nodeContains(Node container, Node contained, boolean ignor
             if (bl>contained.getBeginLine()) return false;
             if (bl==contained.getBeginLine() && bc>contained.getBeginColumn()) return false;
             if (container.getEndLine()<contained.getEndLine()) return false;
-            if (container.getEndLine()==contained.getEndLine() && container.getEndColumn()<contained.getEndColumn()) return false;
-            return true;
+            return !(container.getEndLine() == contained.getEndLine() && container.getEndColumn() < contained.getEndColumn());
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/PositionUtils.java:136]: Should it be <=?

File: javaparser-core/src/main/java/com/github/javaparser/ast/Node.java [hunk 2]
```diff
@@ -330,9 +328,7 @@ public void setParentNode(Node parentNode) {
 
     protected void setAsParentNodeOf(List<? extends Node> childNodes) {
         if (childNodes != null) {
-            Iterator<? extends Node> it = childNodes.iterator();
-            while (it.hasNext()) {
-                Node current = it.next();
+            for (Node current : childNodes) {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/ast/Node.java:331]: thank you, my eyes hurt at looking at the old way of doing iterations :)

File: javaparser-core/src/main/java/com/github/javaparser/ast/comments/Comment.java [hunk 3]
```diff
@@ -91,7 +91,7 @@ public void setCommentedNode(Node commentedNode)
     {
         if (commentedNode==null)
         {
-            this.commentedNode = commentedNode;
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/ast/comments/Comment.java:94]: We should also fix the brackets in this file but this change is good

File: javaparser-core/src/main/java/com/github/javaparser/ast/visitor/DumpVisitor.java [hunk 4]
```diff
@@ -1587,8 +1587,7 @@ public void visit(LambdaExpr n, Object arg) {
         printJavaComment(n.getComment(), arg);
 
         List<Parameter> parameters = n.getParameters();
-		boolean printPar = false;
-		printPar = n.isParametersEnclosed();
+		boolean printPar = n.isParametersEnclosed();
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/ast/visitor/DumpVisitor.java]: we could even make printPar final

**Comment by ftomassetti** [comment]
It looks good to me (I did not look at the JavaCC file)

File: javaparser-core/src/main/java/com/github/javaparser/PositionUtils.java (hunk 1 above)
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/PositionUtils.java:136]: Tomorrow I'll write a bunch of tests for Range and Position, I'll pick this up too.

File: javaparser-core/src/main/java/com/github/javaparser/ast/visitor/DumpVisitor.java (hunk 4 above)
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/visitor/DumpVisitor.java]: If I start making everything final that can be final, I will submit another huge patch :)

File: javaparser-core/src/main/java/com/github/javaparser/ast/comments/Comment.java (hunk 3 above)
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/comments/Comment.java:94]: Fixed.

File: javaparser-core/src/main/java/com/github/javaparser/ast/Node.java (hunk 2 above)
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/Node.java:331]: How did we ever live without generics and foreach?

**Pull request merged**